src/utils/validate-handle.ts

```diff
@@ -0,0 +1,51 @@
+import type { Indexer } from '../indexer'
+
+export type ValidateHandleErrorKind =
+  | 'existed'
+  | 'lengthInvalid'
+  | 'charsInvalid'
+
+export type ValidateHandleResult =
```

Comment:
I think we can simplify the result, by returning string only and adding a `valid` state.

I think for SDK we shouldn’t put too detailed message (Consider i18n)

Reply:
We can expose rules as a constant array if needed.

Reply:
Updated https://github.com/Crossbell-Box/crossbell.js/pull/42/commits/e4c3877f1526fcac24cf2c428adc516703c00f06

Reply:
LGTM!